src/transports/hello_zmq_listener.erl

```diff
@@ -107,13 +103,11 @@ handle_cast(_Cast, State) ->
 code_change(_FromVsn, _ToVsn, State) ->
     {ok, State}.
 
-url_for_zmq(URI = #ex_uri{scheme = "zmq-tcp"}) -> ex_uri:encode(URI#ex_uri{scheme = "tcp"});
-url_for_zmq(URI = #ex_uri{scheme = "zmq-ipc"}) -> ex_uri:encode(URI#ex_uri{scheme = "ipc"}).
-
 %% --------------------------------------------------------------------------------
 %% -- helpers
-url_for_log1(URI) ->
-    list_to_binary(ex_uri:encode(URI)).
+default_port(undefined) -> 27000;
+default_port(0) -> 27000;
+default_port(Port)      -> Port.
```

Comment:
Why is the default port logic needed? The whole point of dnss is that we can listen on some random port and publish that to world, instead of having to define the port in advance.....